telemetry/detectors/fly/fly.go

```diff
@@ -0,0 +1,59 @@
+package fly
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.25.0"
+)
+
+var _ resource.Detector = (*detector)(nil)
+
+type detector struct{}
+
+func NewDetector() resource.Detector {
+	return &detector{}
+}
+
+func (d *detector) Detect(_ context.Context) (*resource.Resource, error) {
+	if !isFly() {
+		return resource.Empty(), nil
+	}
+
+	attrs := []attribute.KeyValue{
+		semconv.CloudProviderKey.String("fly"),
+	}
+
+	attrs = addEnvAttr(attrs, "FLY_APP_NAME", attribute.Key("fly.app_name"))
+	attrs = addEnvAttr(attrs, "FLY_IMAGE_REF", attribute.Key("fly.image_ref"))
+	attrs = addEnvAttr(attrs, "FLY_MACHINE_ID",
+		semconv.ServiceInstanceIDKey,
+		attribute.Key("fly.machine_id"),
+	)
+	attrs = addEnvAttr(attrs, "FLY_MACHINE_VERSION", "fly.machine_version")
```

Comment:
```suggestion
	attrs = addEnvAttr(attrs, "FLY_MACHINE_VERSION", attribute.Key("fly.machine_version"))
```

unless I've missed something?

Reply:
Oops, thank you!